dts/bindings/interrupt-controller/interrupt-controller.yaml

```diff
@@ -12,3 +12,6 @@ properties:
     type: int
     required: true
     description: Number of items to expect in an interrupt specifier
+  "has-multi-instance":
```

Comment:
This is needed for the DT macro in `sw_isr_common.c` to get the device pointer of a root interrupt controller.
Alternatively this might be replaceable by Kconfigs (`INTC_HAS_MULTI_INSTANCE`, selected by interrupt controller that supports multi-instance, but not necessarily enabled) to only compile the multi-instance related code when `INTC_ENABLE_MULTI_INSTANCE` is enabled

Reply:
Consistent with Zephyr's driver model, it's better to leave the gate assuming that drivers can be multi-instance. That's also a DT thing, and not a Kconfig thing. The number of instances of a dt-compat is implicitly part of DT. So I would say that it's better to proceed under the assumption that this is always `true`. There is no real need to reiterate something explicitly that is implicit in devicetree.

If this is part of the `sw_isr_common.c`, please make the necessary adjustments to the other PR.

Reply:
this prop is removed in #63172